provider/groupmembership/group_membership_resource.go

```diff
@@ -0,0 +1,234 @@
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package groupmembership
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	sdk "github.com/openshift-online/ocm-sdk-go"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+)
+
+type GroupMembershipResource struct {
+	collection *cmv1.ClustersClient
+}
+
+var _ resource.ResourceWithConfigure = &GroupMembershipResource{}
+var _ resource.ResourceWithImportState = &GroupMembershipResource{}
+
+func New() resource.Resource {
+	return &GroupMembershipResource{}
+}
+
+func (g *GroupMembershipResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_group_membership"
+}
+
+func (g *GroupMembershipResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages user group membership.",
+		Attributes: map[string]schema.Attribute{
+			"cluster": schema.StringAttribute{
+				Description: "Identifier of the cluster.",
+				Required:    true,
+			},
+			"group": schema.StringAttribute{
+				Description: "Identifier of the group.",
+				Required:    true,
+			},
+			"id": schema.StringAttribute{
+				Description: "Identifier of the membership.",
+				Computed:    true,
+			},
+			"user": schema.StringAttribute{
+				Description: "user name.",
+				Required:    true,
+			},
+		},
+	}
+	return
+}
+
+func (g *GroupMembershipResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	connection, ok := req.ProviderData.(*sdk.Connection)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *sdk.Connaction, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	g.collection = connection.ClustersMgmt().V1().Clusters()
+}
+
+func (g *GroupMembershipResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Get the plan:
+	state := &GroupMembershipState{}
+	diags := req.Plan.Get(ctx, state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Wait till the cluster is ready:
+	resource := g.collection.Cluster(state.Cluster.String())
```

Comment:
When accessing a TF variable, you should use the Value<Type>() method (i.e., `state.Cluster.ValueString()`)
Other than that, this PR lgtm.